storage/redis/peer_store.go

```diff
@@ -267,28 +238,77 @@ type peerStore struct {
 	wg     sync.WaitGroup
 }
 
-func (ps *peerStore) groups() []string {
-	return []string{bittorrent.IPv4.String(), bittorrent.IPv6.String()}
-}
+var addressFamilies = [2]string{"4", "6"}
 
-func (ps *peerStore) leecherInfohashKey(af, ih string) string {
-	return af + "_L_" + ih
-}
+// Key that stores the peers for an InfoHash.
+//
+// "4S0102030405060708090a0b0c0d0e0f1011121314" => IPv4 Seeder with Peer ID 0102030405060708090a0b0c0d0e0f1011121314
+// "6L0204090405060708060a0b0c0d0e0f1011121328" => IPv6 Seeder with Peer ID 0204090405060708060a0b0c0d0e0f1011121328
+func peerKey(infoHash bittorrent.InfoHash, seed bool, ip netaddr.IP) string {
+	var b strings.Builder
+	switch {
+	case ip.Is4(), ip.Is4in6():
+		b.WriteString("4")
+	case ip.Is6():
+		b.WriteString("6")
+	default:
+		panic("unknown IP version")
+	}
 
-func (ps *peerStore) seederInfohashKey(af, ih string) string {
-	return af + "_S_" + ih
-}
+	if seed {
+		b.WriteString("S")
+	} else {
+		b.WriteString("L")
+	}
 
-func (ps *peerStore) infohashCountKey(af string) string {
-	return af + "_infohash_count"
+	b.WriteString(infoHash.String())
+	return b.String()
 }
 
-func (ps *peerStore) seederCountKey(af string) string {
-	return af + "_S_count"
+// Key that stores the cardinality of the peers of an IP version.
+//
+// "4L" => IPv4 Leechers
+// "6S" => IPv6 Seeders
+func peerCountKey(seed bool, ip netaddr.IP) string {
+	var b strings.Builder
+
+	switch {
+	case ip.Is4(), ip.Is4in6():
+		b.WriteString("4")
+	case ip.Is6():
+		b.WriteString("6")
+	default:
+		panic("unknown IP version")
+	}
+
+	if seed {
+		b.WriteString("S")
+	} else {
+		b.WriteString("L")
+	}
+
+	return b.String()
 }
 
-func (ps *peerStore) leecherCountKey(af string) string {
-	return af + "_L_count"
+// Key that stores the total number of infohashes that has peers of an IP
+// version.
+//
+// "I4" => number of IPv4 InfoHashes
+// "I6" => number of IPv6 InfoHashes
+func infohashCountKey(ip netaddr.IP) string {
+	var b strings.Builder
```

Comment:
I wonder if the `Builder` is necessary or actually adds overhead. We know this is gonna be two bytes, so we might as well just allocate a slice for that.